config/config.go

```diff
@@ -449,7 +449,22 @@ func (cfg *Config) newBasicHost(swrm *swarm.Swarm, eventBus event.Bus) (*bhost.B
 // NewNode constructs a new libp2p Host from the Config.
 //
 // This function consumes the config. Do not reuse it (really!).
-func (cfg *Config) NewNode() (host.Host, error) {
+func (cfg *Config) NewNode() (_ host.Host, createErr error) {
+
+	defer func() {
+		if createErr != nil {
```

Comment:
Does this cover everything not already stopped by fx and how does this play with the closing that happens when autonat fails to start?